Add user and dev docs (#45)
streamshub/streamshub-mcp: strimzi-mcp/README.md

# Strimzi MCP Server

A Quarkus application that provides Strimzi Kafka management tools via MCP (Model Context Protocol) for AI assistants and automation.

> [!WARNING]
> This project is in early alpha and under active development. APIs, tool definitions, and configuration may change without notice.

## Overview

The Strimzi MCP Server enables AI assistants to manage and troubleshoot Strimzi-managed Kafka clusters on Kubernetes.
Ask your AI assistant questions like:

- "List all Kafka clusters"
- "Show me error logs from the last hour"
- "Are there any under-replicated partitions?"
- "Diagnose connectivity issues"

The AI assistant uses MCP tools to interact with your Kubernetes cluster and provide answers.

## Quick start

### Prerequisites

- Kubernetes cluster with Strimzi installed
- `kubectl` configured to access your cluster
- Java 21 or later and Maven 3.8 or later (for local development)

> ### Deploy Strimzi (if needed)

Reviewer: ```suggestion
#### Deploy Strimzi (if needed)
```

```bash

> ../dev/scripts/setup-strimzi.sh

Reviewer: ```suggestion
../dev/scripts/setup-strimzi.sh deploy
```

Reviewer: And I wonder if mention other options as well. Like prometheus/loki solo scripts and option in strimzi script to deploy prometheus/loki by these scripts as well.

```

### Run locally

```bash
mvn clean package
mvn quarkus:dev
```

The server starts on `http://localhost:8080/mcp`.

### Connect AI assistant

For Claude Code:

```bash
claude mcp add --transport http strimzi http://localhost:8080/mcp
```

For Claude Desktop, add the following to your configuration file:

```json
{
  "mcpServers": {
    "strimzi": {
      "transport": "http",
      "url": "http://localhost:8080/mcp"
    }
  }
}
```

### Verify

Ask your AI assistant:

- "List all Kafka clusters"
- "What is the status of mcp-cluster?"
- "Diagnose issues with mcp-cluster"

## Documentation

For comprehensive documentation, see [`../docs/strimzi-mcp/`](../docs/strimzi-mcp/):

> - **[Installation](../docs/strimzi-mcp/installation.md)** -- Local and Kubernetes deployment

Reviewer: ```suggestion
- **[Installation](../docs/strimzi-mcp/installation.md)** -- Strimzi MCP deployment locally or on top of Kubernetes
```

Not great wording, but the original seems to be a little bit misleading.

> - **[Configuration](../docs/strimzi-mcp/configuration.md)** -- Environment variables, integrations, security

Reviewer: ```suggestion
- **[Configuration](../docs/strimzi-mcp/configuration.md)** -- Details about configuration options, integrations, security
```

- **[Tools reference](../docs/strimzi-mcp/tools.md)** -- Complete tool catalog with parameters
- **[Usage examples](../docs/strimzi-mcp/usage-examples.md)** -- Practical workflows
- **[Troubleshooting](../docs/strimzi-mcp/troubleshooting.md)** -- Common issues and solutions

## Key features

### Composite diagnostic tools

Multi-step workflows with LLM-guided triage:

- **`diagnose_kafka_cluster`** -- Comprehensive cluster health check
- **`diagnose_kafka_connectivity`** -- Connectivity troubleshooting
- **`diagnose_kafka_metrics`** -- Metrics analysis and anomaly detection
- **`diagnose_operator_metrics`** -- Operator performance analysis

### Cluster management tools

- **List and inspect** -- Clusters, topics, node pools, operators
- **Pod operations** -- Status, logs, events
- **Bootstrap servers** -- Connection endpoints
- **Certificates** -- TLS certificate information

> ### Log collection

Reviewer: maybe add a little bit more details?

- **Error analysis** -- Automatic detection and categorization
- **Advanced filtering** -- By log level, keywords, time ranges
- **Multiple providers** -- Kubernetes API or Loki
- **Progress tracking** -- Real-time updates

> ### Metrics analysis

Reviewer: maybe add a little bit more details?

- **Category-based queries** -- Replication, throughput, performance, resources
- **Interpretation guides** -- Thresholds and diagnostic recommendations
- **Multiple providers** -- Pod scraping or Prometheus
- **Kafka Exporter metrics** -- Consumer lag, topic partitions

### Security guardrails

- **Log redaction** -- Automatic removal of sensitive patterns
- **Response size limits** -- Prevents excessive responses
- **Rate limiting** -- Per-category request throttling

### Resource monitoring

Real-time subscriptions to Kubernetes resources with automatic notifications when state changes.

## Kubernetes deployment

```bash
# Build container image
mvn clean package -DskipTests \
  -Dquarkus.container-image.build=true \
  -Dquarkus.container-image.push=true

# Deploy to Kubernetes
kubectl apply -f install/

# Verify
kubectl -n streamshub-mcp get pods
```

For detailed instructions, see the [installation guide](../docs/strimzi-mcp/installation.md).

> ## Configuration

Reviewer: Maybe a sentece or two that this is an example config for log collector ?
